code/modules/projectiles/guns/energy/special.dm

```diff
@@ -271,6 +271,12 @@
 		return
 	p_orange.link_portal(p_blue)
 	p_blue.link_portal(p_orange)
+	log_game("[src] ([REF(src)]) linked portals between [AREACOORD(p_orange)] and [AREACOORD(p_blue)]")
+
+/obj/item/gun/energy/wormhole_projector/process_fire(atom/target, mob/living/user, message, params, zone_override, bonus_spread)
+	. = ..()
+	if(.) // it fired
+		log_game("[key_name(user)] fired a [src] ([REF(src)]) in [AREACOORD(src)]")

```

Comment:
this is already logged by the projectile's `on_hit()`.

Reply:
This is more about linking the user to the specific portal gun REF()